sklearn/base.py

```diff
@@ -576,21 +576,25 @@ def _validate_data(
         no_val_X = isinstance(X, str) and X == "no_validation"
         no_val_y = y is None or isinstance(y, str) and y == "no_validation"
 
+        if no_val_X and no_val_y:
+            raise ValueError("Validation should be done on X, y or both.")
```

Comment:
probably a dummy test for this wouldn't hurt.

Reply:
Yea, makes sense. I updated the test to check this code path.